### PR TITLE
Feat/create the menu

### LINKED ISSUE #42: Create the Menu
# :white_check_mark: What's needed
Create the Menu (with IGame)

## 📋 Subtasks
- [x] launch at the beginning
- [x] display graphics libs
- [x] display games libs
- [ ] display scores
- [x] username space
- [x] start game

## :file_folder: Files required
src/include/IGame

### PR DESCRIPTION
**Pull request by mathieurobert1**
Create the menu (Game)
Launch games
handle command (next game; go back menu; restart game)
write username

More details : #42

### CODE REVIEW

File: src/Core/Core.cpp [hunk 1]
```diff
@@ -110,12 +111,26 @@ void Arcade::Core::commandExit()
...
+    _gameLib->startGame();
 }
 
 void Arcade::Core::commandNextGame()
 {
-    //implement when we will have a Game
+    if (dynamic_cast<Menu::Menu*>(_gameLib.get()))
+        return;
+    _gameLib->stopGame();
+    _gameLib = nullptr;
```
Comment (Thomaltarix) [inline src/Core/Core.cpp]: You could use the `reset` method from unique pointers

File: src/Core/Menu/menu.cpp [hunk 2]
```diff
@@ -0,0 +1,388 @@
...
+    _userNameDisplay = "_ _ _ _ _ _ _ _ _ _ _ _ _ _ _";
+    _userName = "";
+    _userNameIndex = 0;
+    _selectedGraphic = selectedGraphic;
+    _selectedGame = "";
+}
+
+Arcade::Menu::Menu::~Menu()
+{
+
```
Comment (Thomaltarix) [inline src/Core/Menu/menu.cpp]: Nothing to destroy ?

File: src/Core/Menu/menu.cpp [hunk 3]
```diff
@@ -0,0 +1,388 @@
...
+int Arcade::Menu::Menu::startGame()
+{
+    //Entities creation
+    createEntityBackground();
+    createEntityPlay();
+    createUserName();
+    createLibsButtons();
+    //Text creation
+    createText(_userNameDisplay, 5, 23, 24);
```
Comment (Thomaltarix) [inline src/Core/Menu/menu.cpp]: How does it work using ncurses ? :eyes:

File: src/Core/Menu/menu.cpp [hunk 4]
```diff
@@ -0,0 +1,388 @@
...
+            _selectedGame = _listGame[tmp[1] - 1];
+            _cursor = tmp;
+        }
+        if (_cursor[0] == 3) {
+            _isPlayPressed = true;
+        }
+    }
+}
+
+void Arcade::Menu::Menu::handleArrowKey(int key)
```
Comment (Thomaltarix) [inline src/Core/Menu/menu.cpp]: You could replace this function by a function array

File: src/Core/Menu/menu.hpp [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+#include "IGame.hpp"
+
+#define BUTTON_IDLE "assets/menu/button-idle"
+#define BUTTON_PRESSED "assets/menu/button-pressed"
+#define BUTTON_SELECTED "assets/menu/button-selected"
+
+namespace Arcade {
+    namespace Menu {
+        class Menu : public virtual IGame {
+            private:
```
Comment (Thomaltarix) [inline src/Core/Menu/menu.hpp]: You could put comments to define "groups" of variable

**Review by Thomaltarix — APPROVED** [review]
Seems clean :ok_hand:

File: src/Core/Menu/menu.cpp (hunk 3 above)
Comment (mathieurobert1) [reply src/Core/Menu/menu.cpp]: Not sure for the moment. Waiting a working Ncurses

**Pull request merged**